src/section/section_map.rs

```diff
@@ -281,71 +281,80 @@ impl SectionMap {
     }
 
     /// Returns the known section keys and any recent keys we still hold.
-    pub fn keys(&self) -> impl Iterator<Item = (&Prefix<XorName>, &SectionKeyInfo)> {
+    pub fn keys(&self) -> impl Iterator<Item = (&Prefix<XorName>, &bls::PublicKey)> {
         self.keys
             .iter()
             .chain(self.recent_keys.iter().map(|(p, k)| (p, k)))
     }
 
-    pub fn latest_compatible_key(&self, name: &XorName) -> Option<&SectionKeyInfo> {
+    #[cfg_attr(feature = "mock_base", allow(clippy::trivially_copy_pass_by_ref))]
+    pub fn has_key(&self, key: &bls::PublicKey) -> bool {
+        self.keys.values().any(|known_key| known_key == key)
+    }
+
+    pub fn latest_compatible_key(&self, name: &XorName) -> Option<&bls::PublicKey> {
+        // `keys` is already ordered newest to oldest.
```

Comment:
`keys` is a `BTreeMap` keyed by prefix, why the comment says `is already ordered newest to oldest` ?
or you mean the `keys` function? 
however, the function seems chain self.keys first then the self.recent_keys ?

Reply:
So `keys` contains only the latest (newest) keys for every section and it is then chained with `recent_keys` which has the previous keys in order newest to oldest. So the resulting iterator also yields the keys newest to oldest. The `find` function then returns the first matching entry which also happens to be the newest entry because of that. Maybe the comment should be clearer?

Reply:
Yes, I think the comment shall be clearer.

acutally, according what you explained, there are actually two set of keys chained within the result from `fn keys()`: one is keys of other section, and the other is previous keys (of ours ?).
which, to me, is bit weird/confusing to chain these two together. :(

Reply:
No, it contains only their keys. `keys` are the latest ones we know of and `recent_keys` is a few of previous ones, but still theirs. I'll try to make the comment clearer.